pkg/stanza/adapter/converter.go

```diff
@@ -159,6 +159,9 @@ func (c *Converter) workerLoop() {
 					rl := pLogs.ResourceLogs().AppendEmpty()
 					upsertToMap(e.Resource, rl.Resource().Attributes())
 					sl = rl.ScopeLogs().AppendEmpty()
+					if e.ScopeName != "" {
+						sl.Scope().SetName(e.ScopeName)
+					}
```

Comment:
Unfortunately, I do no think this is the technically correct solution. Say we have four logs, two each from "Resource 1" and "Resource 2", and within each pair, one log is from "Scope A" and the other from "Scope B".

- Resource 1, Scope A: "foo"
- Resource 1, Scope B: "bar"
- Resource 2, Scope A: "hello"
- Resource 2, Scope B: "world"

Current logic which [groups logs by resource](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/19f2a53e42f9ace56e14a624fdaa56a482082cb8/pkg/stanza/adapter/converter.go#L150-L156) but ignores the scope would group these together like this:
- Resource 1: { Scope "": { "foo", "bar" } }
- Resource 2: { Scope "": { "hello", "world" } }

The correct result would group logs according to resource and scope:
- Resource 1: { Scope A: { "foo" }, Scope B: { "bar" } }
- Resource 2: { Scope A: { "hello" }, Scope B: { "world" } }

However, this behavior proposed here would result in the following, because it still only creates one scope per resource (even though it does assign the name).
- Resource 1: { Scope **A**: { "foo", "bar" } }
- Resource 2: { Scope **A**: { "hello", "world" } }

I think the solution must involve managing a set of scopes for each resource, e.g:
```go
// map[resourceID]map[e.ScopeName]scopeIndex
scopeIndexByResource := make(map[uint64]map[string]int)
scopesInThisResource := scopeIndexByResource[resourceID]
scopeIdx, ok := scopesInThisResource[e.ScopeName]
if !ok {
    scopesInThisResource[e.ScopeName] = rl.ScopeLogs().Len()
    sl = rl.ScopeLogs().AppendEmpty()
    sl.SetName(e.ScopeName)
} else {
    sl = rl.ScopeLogs().At(scopeIdx)
}
```